From f9f4dad4ceac84b4ca496c82e1e2824ca986c3e2 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 4 Aug 2010 11:53:00 +0200
Subject: [PATCH] glamo-core: move glamo_engine_reg_set definition outside #ifdef CONFIG_DEBUG_FS

* it's also used outside
* building without CONFIG_DEBUG_FS fails with error: array type has incomplete element type
---
 drivers/mfd/glamo-core.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/mfd/glamo-core.c b/drivers/mfd/glamo-core.c
index 23a0026..468194f 100644
--- a/drivers/mfd/glamo-core.c
+++ b/drivers/mfd/glamo-core.c
@@ -398,12 +398,6 @@ static const struct file_operations debugfs_regs_ops = {
 	.release	= single_release,
 };
 
-struct glamo_engine_reg_set {
-	uint16_t reg;
-	uint16_t mask_suspended;
-	uint16_t mask_enabled;
-};
-
 static void glamo_init_debugfs(struct glamo_core *glamo)
 {
 	glamo->debugfs_dir = debugfs_create_dir("glamo3362", NULL);
@@ -429,6 +423,12 @@ static void glamo_exit_debugfs(struct glamo_core *glamo)
 }
 #endif
 
+struct glamo_engine_reg_set {
+	uint16_t reg;
+	uint16_t mask_suspended;
+	uint16_t mask_enabled;
+};
+
 struct glamo_engine_desc {
 	const char *name;
 	uint16_t hostbus;
-- 
1.7.1.1

